### PR TITLE
[CP-beta]Update Android plugin templates for newer AGP

### PR DESCRIPTION
**Pull request by flutteractionsbot**
This pull request is created by [automatic cherry pick workflow](https://github.com/flutter/flutter/blob/main/docs/releases/Flutter-Cherrypick-Process.md#automatically-creates-a-cherry-pick-request)
Please fill in the form below, and a flutter domain expert will evaluate this cherry pick request.

### Issue Link:
Part of https://github.com/flutter/flutter/issues/156111

### Changelog Description:
Updates the Android plugin templates to use newer configurations that have better compatibility with recent versions of AGP.

### Impact Description:
Without this, plugins created by `flutter create` will have deprecation warnings out of the box for developers whose projects use new versions of AGP, and will eventually fail to compile without manual updates.

### Workaround:
Manually change the Java version after running `flutter create`

### Risk:
What is the risk level of this cherry-pick?

  - [x] Low
  - [ ] Medium
  - [ ] High

### Test Coverage:
Are you confident that your fix is well-tested by automated tests?

  - [ ] Yes
  - [x] No

### Validation Steps:
1. `flutter create -t plugin some_plugin --platforms=android`
2. Update the example project to the latest version of AGP
3. Build

There should not be warnings about `source value 8 is obsolete` or `target value 8 is obsolete`.

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@reidbaker @gmackall Thoughts on cherry-picking this to the new beta? It seems like getting this out sooner rather than later will slightly reduce problems overall, and it's low risk, but it's also something that individual developers could do themselves (if they know to).

**Review by reidbaker — APPROVED** [review]

**Comment by reidbaker** [comment]
> @reidbaker @gmackall Thoughts on cherry-picking this to the new beta? It seems like getting this out sooner rather than later will slightly reduce problems overall, and it's low risk, but it's also something that individual developers could do themselves (if they know to).

I am for this CP

**Comment by gmackall** [comment]
I agree it is low risk and helpful, which is enough reason for me!

**Review by gmackall — APPROVED** [review]

**Comment by reidbaker** [comment]
@gmackall can I get you to reformat the change log description to follow these best practices. 

https://github.com/flutter/flutter/blob/master/docs/releases/Hotfix-Documentation-Best-Practices.md

In the past this was something a release engineer would do but we are moving to having cherry pickers do it since they have a better understanding of the change. 

(or tell me to do it, i can do this one i am just running though the cps this morning.

**Comment by itsjustkevin** [comment]
@stuartmorgan this cherry-pick has failing tests that did not pass on rerun.

**Comment by stuartmorgan** [comment]
@itsjustkevin Those failures don't look related to the PR (e.g., one is that a hot reload test is expecting a very specific success message and what's being output is a different success message). Are any CPs passing right now?

**Comment by itsjustkevin** [comment]
@stuartmorgan looks like Mac tool_integration_tests_5_5 is also failing on https://github.com/flutter/flutter/pull/158131.

**Comment by reidbaker** [comment]
@bkonyi I think was looking at failures on that test.

**Comment by bkonyi** [comment]
The failures on `Mac tool_integration_tests_5_5` seem to be related to this issue: https://github.com/dart-lang/native/issues/1698.

**Comment by mkustermann** [comment]
> The failures on Mac tool_integration_tests_5_5 seem to be related to this issue: https://github.com/dart-lang/native/issues/1698.

The [check failure](https://github.com/flutter/flutter/pull/157100/checks?check_run_id=32553683214) from [build failure](https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20tool_integration_tests_5_5/3289/overview) shows in the [log](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8732087288775205121/+/u/run_test.dart_for_tool_integration_tests_shard_and_subshard_5_5/stdout)
```
07:29 +31 -1: test/integration.shard/isolated/native_assets_test.dart: flutter build apk succeeds without libraries [E]
  Exception: flutter build failed: 1
  Unhandled exception:
  FormatException: No value was provided for required key: out_dir_shared
  #0      MapJsonUtils.get (package:native_assets_cli/src/json_utils.dart:50:7)
  #1      MapJsonUtils.path (package:native_assets_cli/src/json_utils.dart:25:48)
  #2      new HookConfig (package:native_assets_cli/src/config.dart:88:38)
  #3      new BuildConfig (package:native_assets_cli/src/config.dart)
  #4      build (package:native_assets_cli/src/api/build.dart:99:18)
  #5      main (file:///Volumes/Work/s/w/ir/x/t/pZaMJY/package_with_native_assets/hook/build.dart:4:9)
  #6      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:33)
  #7      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
  Building native assets for package:package_with_native_assets failed.
```

The root cause for this is if the test uses newest `package:native_assets_cli` instead of a fixed version (so CI isn't deterministic, tests chan change results if something got uploaded to pub).

The [offending code](https://github.com/flutter/flutter/blob/beta/packages/flutter_tools/test/integration.shard/isolated/native_assets_test.dart#L607)
```dart

  final ProcessResult result2 = await processManager.run(
    <String>[
      flutterBin,
      'pub',
      'add',
      'native_assets_cli',
    ],
    workingDirectory: packageDirectory.path,
  );
  expect(result2, const ProcessResultMatcher());
```

=> This was a problem in the test
=> I believe @matanlurey fix in cc90a424c98698ed78281b007a50cfb1da1c004d resolved this

@reidbaker @bkonyi You may need to cherr-pick cc90a424c98698ed78281b007a50cfb1da1c004d.

If the cherry-picks also upgrade pub dependencies (i.e. updated via `flutter upgrade-packages` ), then you may also want to cherry-pick another fix to native asset tests, namely 31c1292be3fa0b4bac3ed05f035a816ba028f5e1

File: packages/flutter_tools/lib/src/macos/cocoapods.dart [hunk 1]
```diff
@@ -502,6 +502,17 @@ class CocoaPods {
           );
         }
       }
+    } else if (stdout.contains('unknown ISA `PBXFileSystemSynchronizedRootGroup`')) {
+      // CocoaPods does not work with Xcode 16 since it has not yet been
+      // updated to handled synchronized
+      // groups/folders https://github.com/CocoaPods/CocoaPods/issues/12456
+      _logger.printError(
+        'Error: CocoaPods does not support Xcode 16 synchronized groups. '
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/macos/cocoapods.dart:510]: Is this change related to the bug listed? https://github.com/flutter/flutter/issues/156111

Comment (eyebrowsoffire) [reply packages/flutter_tools/lib/src/macos/cocoapods.dart]: This was an issue from switching from the 3.26 branch target to the 3.27 branch target. I ended up doing a manual rebase and force pushed and that seems to have gotten rid of the unwanted changes.
